### PR TITLE
Add text components, update storybook styling

### PR DESCRIPTION
**Pull request by jestapinski**
This PR:

- Adds text components (H1, etc) and adds fonts from mockup to global styles in layouts
- Uses ^ on existing layout pages
- Updates storybook styling and adds text components to it

Screen Shot 2020-02-03 at 10 54 20 AM
Screen Shot 2020-02-03 at 10 54 41 AM

### CODE REVIEW

File: src/components/dev-hub/text/text.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import styled from '@emotion/styled';
+import { fontSize, size } from '../theme';
+
+/**
+ *
+ * @param {Object.string} param
+ *  @param {string} italic should be italic style
+ *  @param {string} bold should be bold weight
+ */
+const variants = ({ bold, collapse }) =>
```
Comment (jestapinski) [inline src/components/dev-hub/text/text.js]: We might also want to pull this out for the time being, what do you think?

File: src/components/dev-hub/layout.js [hunk 2]
```diff
@@ -8,6 +8,15 @@ const globalStyles = css`
   body {
     background: ${colorMap.devBlack};
     color: ${colorMap.devWhite};
+    font-family: 'Fira Mono Bold', 'Space Mono Bold', 'Source Code Pro';
```
Comment (jestapinski) [inline src/components/dev-hub/layout.js]: I'm not sure what all of the rules for when these are used vs Akzidenz, so I stuck with just these for now and when the mockups are finalized we can update this.

File: src/components/dev-hub/layout.js [hunk 3]
```diff
@@ -8,6 +8,15 @@ const globalStyles = css`
...
+  }
+  /* Need overrides for guides.css styles */
```
Comment (greggb) [inline src/components/dev-hub/layout.js]: We'll need to update the GATSBY_SITE name to get rid of that stylesheet. Might also need to work with Sophie to fix the ternary that will load an alternate stylesheet in `html.js`

File: src/components/dev-hub/text/text.js (hunk 1 above)
Comment (greggb) [reply src/components/dev-hub/text/text.js]: fine with me

File: src/components/dev-hub/layout.js (hunk 3 above)
Comment (jestapinski) [reply src/components/dev-hub/layout.js]: Noted

**Pull request merged**